### PR TITLE
Added contactable to tracks

### PR DESCRIPTION
**Pull request by rishabhptr**
Related to #2528 

I have tried couple of different ways to use the contact model for social media fields. I have chosem this approach so as to enable for the user to add all fields at the same time in a single form while creating the track using nested attributes.

Also, as `contact` model has a `has_one` relationship with `conference` model I need to pass the `conference_id` as it is mandatory but I believe this will break the previous mechanism where a conference had one contact which could be fetched to show on the splashpage. _**So we would need to modify this relationship as well.**_

At this point I need to decide if I should use this model or just add the social fields in the main Track/Booth model itself.

image

Apart from this other fields like :code_of_conduct_accepted, :additional_requirements etc are pretty straightforward and after this I can work on the different views related to these details.

### CODE REVIEW

**Comment by differentreality** [comment]
Please do correct me if I am wrong, but I don't think we are providing an option for people to submit information such as the project's source code location or the project's license.

**Comment by rishabhptr** [comment]
@differentreality The Social Media fields covers the different platforms and yes there is no specific fields for the code repository. I intended to have a **website_url**, **email** , **github/repo** fields in the track model itself and use the contact model for capture the media platforms.

But I liked the idea for project license, but should we add a special field for this?

The other information apart from these in the track submission that I propose are :
**:code_of_conduct_accepted**   `boolean`
**:additional_requirements**         `string`
**size**                                          `integer`

**:track_type**   - which can be selected from a drop down with choices much like event_types which I will cover in my next PR.

These are the details that I am planning to include in the submission from the users.

File: db/schema.rb [hunk 1]
```diff
@@ -10,100 +10,103 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20181113195810) do
+ActiveRecord::Schema.define(version: 20190612161235) do
+
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
```
Comment (Ana06) [inline db/schema.rb]: Why is this here? In general, why does this file have that many changes?

Comment (rishabhptr) [reply db/schema.rb]: I think some unwanted changes were introduced during db migrations . I will check it again in my new PR against the GSoC19 branch.

**Comment by differentreality** [comment]
@rishabhptr so what are you suggesting for the other links we want to show (email, license, etc)? Add fields in Track model? And then in Booth model?

**Comment by rishabhptr** [comment]
@differentreality As of now yes, any specific field like email, website/license which we can identify would go in the track/booth model itself.

**Comment by Ana06** [comment]
@differentreality can we get this merged or are there still concerns?

@rishabh92 can you please rebase? there are some commits that shouldn't be here.

We should fix https://github.com/openSUSE/osem/issues/2540 because is blocking the PRs as well.

**Comment by differentreality** [comment]
@rishabhptr please keep an eye out on Travis and fix the linters failures, we cannot merge ortherwise!

**Comment by rishabhptr** [comment]
@differentreality I have fixed the linters, but the feature test were failing due to the webdriver issue which is now resolved. Maybe we should bring in the latest changes to the GSoC19 branch?

**Comment by differentreality** [comment]
> @differentreality can we get this merged or are there still concerns?

@Ana06 not concerns per se, but we will probably need to use the same fields for Booth as well.

**Comment by differentreality** [comment]
Screenshots would be helpful here too :wink:

**Comment by differentreality** [comment]
@rishabhptr please rebase

**Comment by rishabhptr** [comment]
@differentreality Yes, the form is quite longer now. Maybe we could ask for the other fields at a later stage.

Right now when the track is accepted and then confirmed by the submitter, the user is able to manage his track but he cannot edit the details as it make sense to not change them after submission.

However, we can include the option to add fields such as social media in a separate tab in the show page. So that the user can update these details there.

Screenshot from 2019-08-09 18-39-40
Screenshot from 2019-08-09 18-38-46
